plugins/org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/serializer/SemanticSequencerUtil.java

```diff
@@ -116,25 +117,24 @@ public Grammar getSuperGrammar(Grammar grammar) {
 		return result;
 	}
 
-	public boolean canGenerate(IConstraint constraint) {
-		if (constraint.getAssignments().length != constraint.getFeatures().length)
-			return false;
-		IConstraintElement body = constraint.getBody();
-		if (body == null)
-			return false;
-		if (body.getFeatureInfo() != null && !body.isOptional() && !body.isMany()
-				&& !body.getFeatureInfo().getFeature().isMany())
-			return true;
-		if (body.getType() == ConstraintElementType.GROUP) {
-			if (body.getGrammarElement() instanceof UnorderedGroup)
-				return false;
-			for (IConstraintElement child : body.getChildren())
-				if (child.getFeatureInfo() == null || child.isOptional() || child.isMany()
-						|| child.getFeatureInfo().getFeature().isMany())
-					return false;
-			return true;
+	public List<ISemState> getLinearListOfMandatoryAssignments(IConstraint constraint) {
+		Nfa<ISemState> nfa = constraint.getNfa();
+		ISemState state = nfa.getStart();
+		List<ISemState> result = Lists.newArrayList();
+		Set<EStructuralFeature> feature = Sets.newHashSet();
+		while (state != null) {
+			if (state == nfa.getStop())
+				return result.isEmpty() ? null : result;
+			if (state.getFollowers().size() != 1)
+				return null;
+			if (!feature.add(state.getFeature()))
+				return null;
+			if (state != nfa.getStart()) {
+				result.add(state);
+			}
+			state = state.getFollowers().get(0);
 		}
-		return false;
+		return null;
```

Comment:
yes. When null is returned, the SerializerFragment generates and invocation of the BacktrackingSemanticSequencers. When an empty list is returned, it generates code that consumes the values of zero features.